tensilelite/Tensile/Source/lib/include/Tensile/PredictionLibrary.hpp

```diff
@@ -0,0 +1,222 @@
+            SolutionSet<MySolution> rv;
+            if(searchType == SolutionLibrarySearchType::DEFAULT)
+                return rv;
+
+            for(auto const& row : this->solutionmap)
+            {
+                if(debug)
+                    std::cout << row.second->description() << std::endl;
+                rv.insert(row.second);
+            }
+
+            return rv;
+        }
+
+        virtual SolutionVector<MySolution> findTopSolutions(MyProblem const& problem,
+                                                            Hardware const&  hardware,
+                                                            int numSolutions) const override
+        {
+            SolutionVector<MySolution> rv;
+            size_t                     m     = 1;
+            size_t                     n     = 1;
+            size_t                     k     = 1;
+            size_t                     batch = 1;
+            for(size_t i = 0; i < problem.freeIndicesA().size(); i++)
+            {
+                m *= problem.freeSizeA(i);
+            }
+            for(size_t i = 0; i < problem.freeIndicesB().size(); i++)
+            {
+                n *= problem.freeSizeB(i);
+            }
+            for(size_t i = 0; i < problem.boundIndices().size(); ++i)
+            {
+                k *= problem.boundSize(i);
+            }
+            for(size_t i = 0; i < problem.batchIndices().size(); ++i)
+            {
+                batch *= problem.batchSize(i);
+            }
+
+            bool                  debug   = Debug::Instance().printPropertyEvaluation();
+            hip::HipAMDGPU const* pAMDGPU = dynamic_cast<hip::HipAMDGPU const*>(&hardware);
+            size_t                elementSizeA_bits
+                = problem.a().elementBytes() * 8; // TODO update for A/B different types
+            size_t elementSizeB_bits
+                = problem.b().elementBytes() * 8; // TODO update for A/B different types
+            size_t elementSizeC_bits
+                = problem.c().elementBytes() * 8; // TODO update for A/B different types
+            const analytical::Hardware& analaytical_hardware = *(pAMDGPU->analyticalHardware);
+            int                         WGM
+                = std::sqrt(std::floor(analaytical_hardware.N_CU / analaytical_hardware.NUM_XCD));
+            auto selected_tiles = analytical::select_best_macro_tile_size(
+                m,
+                n,
+                k,
+                batch,
+                problem.transA(),
+                problem.transB(),
+                *(pAMDGPU->analyticalHardware),
+                tile_list,
+                elementSizeA_bits,
+                elementSizeA_bits,
+                elementSizeC_bits,
+                0, //mx_block_size -> MX Data types come from rocroller.
+                0.8,
+                debug,
+                false,
+                6);
```

Comment:
@ryanswann-amd , I think you meant to use this here?

```suggestion
                WGM);
```